sharing/handler.go

```diff
@@ -334,3 +390,139 @@ func makeLocation(base *url.URL, reqPath string) string {
 	location.Path = path.Join(location.Path, reqPath)
 	return location.String()
 }
+
+// handleHostingService deals with linksharing via custom URLs.
+func (handler *Handler) handleHostingService(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	host, _, err := net.SplitHostPort(r.Host)
+	if err != nil && strings.Contains(err.Error(), "missing port in address") {
+		host = r.Host
+	} else if err != nil {
+		handler.log.Error("unable to handle request", zap.Error(err))
+		http.Error(w, "unable to handle request", http.StatusInternalServerError)
+		return err
+	}
+
+	access, root, err := handler.getRootAndAccess(host)
+	if err != nil {
+		handler.log.Error("unable to handle request", zap.Error(err))
+		http.Error(w, "unable to handle request", http.StatusInternalServerError)
```

Comment:
I think we should be more specific here, otherwise the people setting up dns entries don't have any information.